x-pack/plugin/rank-vectors/src/main/java/org/elasticsearch/xpack/rank/vectors/mapper/RankVectorsFieldMapper.java

```diff
@@ -180,7 +180,17 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
             return new ArraySourceValueFetcher(name(), context) {
                 @Override
                 protected Object parseSourceValue(Object value) {
-                    return value;
+                    List<?> outerList = (List<?>) value;
```

Comment:
I understand, but I think we will need to handle things other than floats. I will check this out and play around.